### PR TITLE
chore: clean up `debug` statements in preparation for 10.0 release, add `debug` docs

### PR DESCRIPTION
**Pull request by flotwig**
### User facing changelog

If we want to consider this user-facing:

* Removed some verbose `DEBUG=cypress:*` lines and moved others to the `DEBUG=cypress-verbose:*` namespace.

### Additional details

- I noticed that we have added a lot of debug lines to our already-bloated `DEBUG` namespaces.
- With 10.0, there are going to be a lot of changes and potentially a lot of bug reports, so it's important that we take the opportunity to clean up `DEBUG` and make it easier to read and more useful before we start getting reports with `DEBUG` logs in them.
- This PR:
  - Documents the way that `debug` namespaces are handled in `cypress`. Previously, the way we named debug namespaces was not documented.
  - Removes or moves to `cypress-verbose` a lot of `debug` statements, some added in 10.0, some existing in 9.x
    - Of note, `packages/launcher` is now using namespaces instead of just `cypress:launcher`
  - Updates some namespaces to follow the written standard

### How has the user experience changed?

`debug` output is easier to parse and make sense of.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: guides/debug-logs.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Debug Logs
+
+Many Cypress packages use the [`debug` module][debug] to log runtime info to the console.
+
+## Choosing a namespace
+
+The naming scheme for debug namespaces should generally follow this pattern:
+
+```
+cypress:{packageName}:{relative path to file from src root, using : to separate directories, minus index if applicable}
```
Comment (tgriesser) [inline guides/debug-logs.md:10]: I wonder if in the future we can do this automatically, similar to what we do [in the dashboard](https://github.com/cypress-io/cypress-services/blob/9d8899e5c68f2221f81b0a5a79bf1e659d313d71/packages/debug/src/debug.ts#L21-L37)

```
import { makeDebug } from '@packages/debug'
const debug = makeDebug(__filename)
```

File: packages/data-context/src/sources/GitDataSource.ts [hunk 2]
```diff
@@ -326,16 +326,14 @@ export class GitDataSource {
     const stdout = result.stdout.split('\n')
 
     if (result.exitCode !== 0) {
-      debug(`error... stderr`, result.stderr)
+      debug(`error... %o`, result)
```
Comment (MuazOthman) [inline packages/data-context/src/sources/GitDataSource.ts:329]:
```suggestion
      debug(`command execution error: %o`, result)
```

File: packages/data-context/src/sources/GitDataSource.ts [hunk 3]
```diff
@@ -326,16 +326,14 @@ export class GitDataSource {
...
     }
 
     if (stdout.length !== absolutePaths.length) {
-      debug('error... stdout:', stdout)
+      debug('error... %o', result)
```
Comment (MuazOthman) [inline packages/data-context/src/sources/GitDataSource.ts:333]:
```suggestion
      debug('unexpected command execution result: %o', result)
```

**Review by MuazOthman — COMMENTED** [review]
It looks much cleaner now. Thanks!

File: guides/debug-logs.md (hunk 1 above)
Comment (flotwig) [reply guides/debug-logs.md:10]: Yeah, I would love to do this in this repo too so the burden isn't on individual contributors and reviewers to know these guidelines.

**Comment by tgriesser** [comment]
> Do you think this utility will remain long-term?

@flotwig I think we should remove it, it gives such a silly amount of logs when you do `DEBUG=cypress*` that it should probably be removed. @ryanthemanuel I saw you had added the comment for it, have you found any use in it with anything you've needed to work on?

**Comment by ryanthemanuel** [comment]
> > Do you think this utility will remain long-term?
> 
> @flotwig I think we should remove it, it gives such a silly amount of logs when you do `DEBUG=cypress*` that it should probably be removed. @ryanthemanuel I saw you had added the comment for it, have you found any use in it with anything you've needed to work on?

I actually used it a fair amount when I was trying to debug a few race condition type issues with the lifecycle manager. Maybe we can namespace it even further so that it doesn't show up with `cypress*`, something like `private-cypress-trace` or something?

**Pull request merged**